reliabot/reliabot.py

```diff
@@ -42,10 +42,9 @@
 from typing import TextIO
 from typing import Union
 
-from ruamel.yaml import YAML  # ruamel.yaml preserves comments, PyYAML doesn't.
-from ruamel.yaml.comments import CommentedMap
-from ruamel.yaml.comments import CommentedSeq
-from ruamel.yaml.parser import ParserError
+from ruamel.yaml.comments import CommentedMap  # no-reorder
+from ruamel.yaml.comments import CommentedSeq  # no-reorder
+from ruamel.yaml.parser import ParserError  # no-reorder
```

Comment:
I think anything after the first `# no-reorder` is not hoisted. So the extra comments are probably not needed. But these imports should be moved back down below the `try` block that imports `ruamel.yaml`.

I think that may be the last fix needed. I appreciate your keeping at this. The pre-commit checking can be annoying sometimes, but it also catches a lot of errors, preventing bugs and enforcing consistent style.

Reply:
I was employed at one point, I know how much headache ci can cause and prevent 😂 I'll keep bashing at this

> But these imports should be moved back down below the try block that imports ruamel.yaml
Did you mean the try/except block? ci keeps hoisting
    
I did see https://github.com/asottile/reorder-python-imports/blob/3fe08c24a7649551e4f1f77399a233e89e598dec/reorder_python_imports.py#L67 which uses `#noreorder` on `main`

Reply:
Yes, down between lines 151 and 152. The code you linked to definitely seems to use `noreorder` rather than `no-reorder` as the docs have it. Try moving the `from ruamel.yaml import` lines back down with comments on each line. Hopefully something will stick.

Reply:
I do not believe ` # noreorder` works as desired - I think we want `python-reorder-imports` to skip the lines instead of skipping the reordering. Otherwise, it's bugged.

see: https://github.com/dupuy/reliabot/pull/18/commits/a43a20dd510c996e45463b5a6d3420dfdd636049

I solutioned this to use the conditional import from a previous commit - its not as pretty, but I'll raise an issue with the maintainer to clarify the behavior when I get the time.

Reply:
👍